source/initial_temperature/patch_on_S40RTS.cc

```diff
@@ -0,0 +1,206 @@
+/*
+  Copyright (C) 2011 - 2017 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/global.h>
+#include <aspect/adiabatic_conditions/interface.h>
+#include <aspect/initial_temperature/patch_on_S40RTS.h>
+#include <aspect/utilities.h>
+
+namespace aspect
+{
+  namespace InitialTemperature
+  {
+    template <int dim>
+    PatchOnS40RTS<dim>::PatchOnS40RTS ()
+    {}
+
+
+    template <int dim>
+    void
+    PatchOnS40RTS<dim>::initialize ()
+    {
+      Utilities::AsciiDataInitial<dim>::initialize(1);
+      if (s40rts.vs_to_density_method == s40rts.file)
+        {
+          s40rts.profile.initialize(this->get_mpi_communicator());
+          s40rts.vs_to_density_index = s40rts.profile.get_column_index_from_name("vs_to_density");
+        }
```

Comment:
From what I see of S40RTS in the master branch, the code in the `if` statement above should have no effect. The same code is executed in `S40RTSPerturbation::initialize`, which is called by `S40RTSPerturbation::parse_parameters` (weird, but that is what it is at the moment), which is called by `PatchOnS40RTS::parse_parameters`, i.e. before we get to this particular line. In other words: Everything should be ready to go, we just repeat these lines here.